internal/test/assert.go

```diff
@@ -0,0 +1,62 @@
+package test
+
+import (
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type checkerOptions struct {
+	bodyChecker func(*testing.T, []byte) bool
+}
+
+type CheckerOption func(*checkerOptions)
+
+func WithBodyChecker(bodyChecker func(*testing.T, []byte) bool) CheckerOption {
+	return func(opts *checkerOptions) {
+		opts.bodyChecker = bodyChecker
+	}
+}
+
+// EventuallyExpectedStatusCodeAndBody is a helper function that retries the request until
+// it gets the expected status. It also checks the body of the response if a body checker
+// is provided. Default is to retry for 1 minute with 1 second interval.
+func EventuallyExpectedStatusCodeAndBody(
+	t *testing.T, httpClient *http.Client, req *http.Request, expectedStatusCode int, opts ...CheckerOption,
```

Comment:
Instead doing this, how about we make this a generic `http.Response` option + add explicit note about only one option to be able to consume the body. WDYT about that?

Reply:
+ add an option checking status code.